plugins/woocommerce/includes/class-wc-emails.php

```diff
@@ -592,6 +596,72 @@ public function email_addresses( $order, $sent_to_admin = false, $plain_text = f
 		}
 	}
 
+	/**
+	 * Renders any additional fields captured during block based checkout.
+	 *
+	 * @param WC_Order $order         Order instance.
+	 * @param bool     $sent_to_admin If should sent to admin.
+	 * @param bool     $plain_text    If is plain text email.
+	 */
+	public function additional_checkout_fields( $order, $sent_to_admin = false, $plain_text = false ) {
+		if ( ! is_a( $order, 'WC_Order' ) ) {
+			return;
+		}
+
+		$checkout_fields = Package::container()->get( CheckoutFields::class );
+		$fields          = array_merge(
+			$checkout_fields->get_order_additional_fields_with_values( $order, 'contact' ),
+			$checkout_fields->get_order_additional_fields_with_values( $order, 'additional' )
+		);
+
+		if ( ! $fields ) {
+			return;
+		}
+
+		if ( $plain_text ) {
+			echo "\n" . esc_html( wc_strtoupper( __( 'Additional information', 'woocommerce' ) ) ) . "\n\n";
+			foreach ( $fields as $field ) {
+				printf( "%s: %s\n", wp_kses_post( $field['label'] ), wp_kses_post( $field['value'] ) );
+			}
+		} else {
+			echo '<h2>' . esc_html__( 'Additional information', 'woocommerce' ) . '</h2>';
+			echo '<ul class="additional-fields" style="margin-bottom: 40px;">';
+			foreach ( $fields as $field ) {
+				printf( '<li><strong>%s</strong>: %s</li>', wp_kses_post( $field['label'] ), wp_kses_post( $field['value'] ) );
+			}
+			echo '</ul>';
+		}
+	}
+
+	/**
+	 * Renders any additional address fields captured during block based checkout.
+	 *
+	 * @param string   $address_type Address type.
+	 * @param WC_Order $order         Order instance.
+	 * @param bool     $sent_to_admin If should sent to admin.
+	 * @param bool     $plain_text    If is plain text email.
+	 */
+	public function additional_address_fields( $address_type, $order, $sent_to_admin = false, $plain_text = false ) {
+		if ( ! is_a( $order, 'WC_Order' ) ) {
+			return;
+		}
+
+		$checkout_fields = Package::container()->get( CheckoutFields::class );
+		$fields          = $checkout_fields->get_order_additional_fields_with_values( $order, 'address', $address_type );
+
+		if ( ! $fields ) {
+			return;
+		}
+
+		foreach ( $fields as $field ) {
+			if ( $plain_text ) {
+				printf( "%s: %s\n", wp_kses_post( $field['label'] ), wp_kses_post( $field['value'] ) );
+			} else {
+				printf( '<br><strong>%s</strong>: %s', wp_kses_post( $field['label'] ), wp_kses_post( $field['value'] ) );
```

Comment:
I'm just wondering, is it ever possible that a field will get here without a value? I see we're checking for `if ( '' === $value ) {` in `get_order_additional_fields_with_values` but is there _ever_ a case where it would be null and this could cause a warning? (same goes for the use in `additional_checkout_fields`)

Reply:
https://github.com/woocommerce/woocommerce/pull/43519/commits/1fff91f175892790b4037c12ed50dee4cb35dd92